### ISSUE TITLE
Best practices for linking with third-party libraries on windows

### ISSUE THREAD

**Issue by mattip** [mesonbuild/meson-python#268]
I understand meson prefers to use pkg-config as the standard discovery method for third-party packages. Is there a recommended way to use pkg-config on windows? In numpy/numpy#23027 I tried to get pkg-config via `choco install pkgconfiglite` but it seems [the package is not properly signed](https://github.com/numpy/numpy/actions/runs/3947067063/jobs/6755464001#step:5:38). Before I go down the rabbit-hole of getting this to work, has someone already done this?

**Comment by rgommers** [comment]
In SciPy I think we're getting `pkg-config` from `choco install rtools` (see https://github.com/scipy/scipy/blob/main/.github/workflows/windows.yml#L41). Conda-forge also has Windows packages: https://anaconda.org/conda-forge/pkg-config. So that's at least two good options. No idea about whether or not to trust `pkgconfiglite` or not.

Also note that you need this only for BLAS/LAPACK right now, and I'll implement a full equivalent of `numpy.distutils` manual scanning for those libraries soon (hopefully in time for Meson 1.1.0).

**Comment by mattip** [comment]
ahh, cool. So there will be some changes needed once that is done, and the current work does not have to be "best practices". I can just copy what SciPy has for windows.

**Comment by rgommers** [comment]
Indeed. I think for that particular use case where there is no distributor who has taken care of properly configured pkg-config & relocatable builds, we may end up using a `native-file.ini` (like numpy's `site.cfg`) to point directly to the shared library directory, and be done with it.

The question is answered, so I'll close this. Please feel free to add more questions of course if it helps.
